RichString.c

```diff
@@ -59,7 +59,7 @@ static void RichString_setLen(RichString* this, int len) {
 }
 
 void RichString_rewind(RichString* this, int count) {
-   RichString_setLen(this, this->chlen - count);
+   RichString_setLen(this, this->chlen > count ? this->chlen - count : 0);
```

Comment:
Note to self: Once the types of `RichString.chlen` and `count` are changed to unsigned, we can change this expression to a `saturatingSub()` call to make the expression more concise to read. However, I found missed optimizations with this: In GCC (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=118679) and in Clang
(https://github.com/llvm/llvm-project/issues/124714)